.github/workflows/undeploy-1-start.yml

```
@@ -63,9 +64,18 @@ jobs:
             ${{ secrets.HOST }}
             ${{ secrets.HOST_DATA }}
 
-      - name: Undeploy
+      - name: Undeploy spack-packages
+        # ssh into deployment environment, remove all the unneeded spack-packages installations
+        run: |
+          ssh ${{ secrets.USER}}@${{ secrets.HOST }} -i ${{ steps.ssh.outputs.private-key-path }} /bin/bash <<EOT
```

```suggestion
          ssh ${{ secrets.USER}}@${{ secrets.HOST }} -i ${{ steps.ssh.outputs.private-key-path }} /bin/bash <<'EOT'
```
I don't think you need any `bash` expansion within the heredoc script, so I would escape the whole script